third_party/wabt/src/interp/jit/jit-backend.cc

```diff
@@ -216,6 +216,50 @@ static void emitBinary(sljit_compiler* compiler, Instruction* instr) {
                  args[0].argw, args[1].arg, args[1].argw);
 }
 
+static void emitUnary(sljit_compiler* compiler, Instruction* instr) {
+  Operand* operand = instr->operands();
+  JITArg args[2];
+
+  for (int i = 0; i < 2; ++i) {
+    operandToArg(operand + i, args[i]);
+  }
+
+  sljit_s32 opcode;
+
+  switch (instr->opcode()) {
+    case Opcode::I32Ctz:
+      opcode = SLJIT_CTZ32;
+      break;
+    case Opcode::I32Clz:
+      opcode = SLJIT_CLZ32;
+      break;
+    case Opcode::I64Ctz:
+      opcode = SLJIT_CTZ;
+      break;
+    case Opcode::I64Clz:
+      opcode = SLJIT_CLZ;
+      break;
+    case Opcode::I32Popcnt:
+    case Opcode::I64Popcnt:
+      // Not supported yet.
+      return;
+    default:
+      WABT_UNREACHABLE;
+      break;
+  }
+
+  // If the operand is an immidiate then it is necesarry to move it into a
+  // register.
+  if (args[0].arg & SLJIT_IMM) {
+    sljit_emit_op1(compiler, SLJIT_MOV, SLJIT_R0, 0, args[0].arg, args[0].argw);
+    sljit_emit_op1(compiler, opcode, args[1].arg, args[1].argw, SLJIT_R0, 0);
```

Comment:
Trick:
set `args[1].arg = SLJIT_R0` and `args[1].argw = 0` and use the generic case. Less code is generated then.